includes/handler/class-quote-request.php

```diff
@@ -0,0 +1,235 @@
+				$follower = Remote_Actors::get_by_uri( object_to_uri( $activity['actor'] ) );
+				if ( ! \is_wp_error( $follower ) && Followers::follows( $follower->ID, $user_id ) ) {
+					self::queue_accept( $activity, $user_id, $post_id );
+				} else {
+					self::queue_reject( $activity, $user_id );
+					$state = false;
+				}
+				break;
+			case ACTIVITYPUB_INTERACTION_POLICY_ANYONE:
+			default:
+				self::queue_accept( $activity, $user_id, $post_id );
+				break;
+		}
+
+		/**
+		 * Fires after an ActivityPub QuoteRequest activity has been handled.
+		 *
+		 * @param array $activity The ActivityPub activity data.
+		 * @param int   $user_id  The local user ID.
+		 * @param bool  $success  True on success, false otherwise.
+		 * @param null  $result   Reserved for future use. Currently always null for QuoteRequest activities.
+		 */
+		\do_action( 'activitypub_handled_quoterequest', $activity, $user_id, $state, null );
+	}
+
+	/**
+	 * ActivityPub inbox disallowed activity.
+	 *
+	 * @param array  $activity The activity array.
+	 * @param null   $user_id  The user ID.
+	 * @param string $type     The type of the activity.
+	 */
+	public static function handle_blocked_request( $activity, $user_id, $type ) {
+		if ( 'quoterequest' !== \strtolower( $type ) ) {
+			return;
+		}
+
+		self::queue_reject( $activity, $user_id );
+	}
+
+	/**
+	 * Send an Accept activity in response to the QuoteRequest.
+	 *
+	 * @see https://codeberg.org/fediverse/fep/src/branch/main/fep/044f/fep-044f.md#example-accept
+	 *
+	 * @param array $activity_object The activity object.
+	 * @param int   $user_id         The user ID.
+	 * @param int   $post_id         The post ID.
+	 */
+	public static function queue_accept( $activity_object, $user_id, $post_id ) {
+		$actor = Actors::get_by_id( $user_id );
+
+		if ( \is_wp_error( $actor ) ) {
+			return;
+		}
+
+		$activity_object['instrument'] = object_to_uri( $activity_object['instrument'] );
+
+		$post_meta = \get_post_meta( $post_id, '_activitypub_quoted_by', false );
+		if ( in_array( $activity_object['instrument'], $post_meta, true ) ) {
+			global $wpdb;
+
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+			$meta_id = $wpdb->get_var(
+				$wpdb->prepare(
+					"SELECT meta_id FROM {$wpdb->postmeta} WHERE post_id = %d AND meta_key = %s AND meta_value = %s LIMIT 1",
+					$post_id,
+					'_activitypub_quoted_by',
+					$activity_object['instrument']
+				)
+			);
+		} else {
+			$meta_id = \add_post_meta( $post_id, '_activitypub_quoted_by', $activity_object['instrument'] );
+		}
+
+		// Only send minimal data.
+		$activity_object = array_intersect_key(
+			$activity_object,
+			array(
+				'id'         => 1,
+				'type'       => 1,
+				'actor'      => 1,
+				'object'     => 1,
+				'instrument' => 1,
+			)
+		);
+
+		$url = \add_query_arg(
+			array(
+				'p'     => $post_id,
+				'stamp' => $meta_id,
+			),
+			\trailingslashit( \home_url() )
```

Comment:
I'm pretty sure `add_query_arg()` handles the correct slashing. Otherwise, it could also be `\home_url( '/' )`